pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -169,7 +185,10 @@ func podStatusToDriverState(podStatus apiv1.PodStatus) v1beta2.DriverState {
 			}
 			return v1beta2.DriverFailedState
 		}
-		return v1beta2.DriverRunningState
+		if getDriverContainerRunningState(podStatus) != nil {
```

Comment:
Looks like all what you need is to tell if all the driver containers are in running state, right? If so, the method can simply return a boolean instead of the state, and the condition becomes more natural.

Reply:
Yeah. `getDriverContainerRunningState` method has replaced by `isDriverContainerRunning` method.